src/hotspot/share/gc/g1/g1YoungCollector.cpp

```diff
@@ -501,13 +501,22 @@ void G1YoungCollector::verify_empty_dirty_card_logs() const {
 }
 #endif // ASSERT
 
-void G1YoungCollector::pre_evacuate_collection_set(G1EvacInfo* evacuation_info, G1ParScanThreadStateSet* per_thread_states) {
+void G1YoungCollector::pre_evacuate_collection_set(G1EvacInfo* evacuation_info) {
+  // Flush early, so later phases don't need to account for per-thread stuff.
```

Comment:
I think this one is duplicate from the comment to `concatenate_dirty_card_logs`.

Reply:
It was intentional, as it's useful to deal with TLABs early for the same reason as the DCQs.  The 2nd line of the comment is about the respective ordering of these two early "flushes".  (The plan of merging these into a single pass over the threads will render the duplicate comment question moot.)